pkg/teeth/github.go

```diff
@@ -0,0 +1,94 @@
+// Copyright 2024 The Authors (see AUTHORS file)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package teeth
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/abcxyz/github-metrics-aggregator/pkg/githubclient"
+	"github.com/google/go-github/v56/github"
+	"github.com/sethvargo/go-retry"
+)
+
+const ()
+
+var (
+	// can be overriden for testing
+	retryMinWaitDuration        = 1 * time.Second
+	retryMaxAttempts     uint64 = 4
+	retryFunc                   = retry.NewFibonacci
+)
+
+// GHConfig contains configuration parameters for the GitHub client.
+type GHConfig struct {
+	Owner string
+	Repo  string
+}
+
+// GitHub is just a wrapper around githubclient.GitHub for functionality specific to teeth.
+// Use NewGitHub to initialize this.
+type GitHub struct {
+	*githubclient.GitHub
+
+	config *GHConfig
+	// can be overriden for testing
+	commentPRFunc func(ctx context.Context, owner, repo string, prNumber int, comment string) (*github.Response, error)
```

Comment:
For fields specific to testing that are embedded on a structure, prefer prefixing with `test`:

```suggestion
	testCommentPRFunc func(ctx context.Context, owner, repo string, prNumber int, comment string) (*github.Response, error)
```